web-services-app/src/main/java/edu/unc/lib/boxc/web/services/processing/SingleUseKeyService.java

```diff
@@ -70,37 +78,67 @@ public boolean keyIsValid(String key) {
     }
 
     /**
-     * Invalidates a key by removing its entry from the CSV
-     * @param key access key of the box-c record
+     * Invalidates a single use key CSV record by removing it from the CSV
+     * @param key access key of the box-c record, may be null for time-based invalidation
      */
     public void invalidate(String key) {
         lock.lock();
         try {
             var csvRecords = parseCsv(CSV_HEADERS, csvPath);
             var updatedRecords = new ArrayList<>();
-            var keyExists = false;
+            var recordExists = false;
             for (CSVRecord record : csvRecords) {
-                if (key.equals(record.get(ACCESS_KEY))) {
-                    keyExists = true;
+                if (recordShouldBeInvalidated(key, record)) {
+                    recordExists = true;
                 } else {
-                    // add the rest of the keys to list
+                    // keep this record as it is valid
                     updatedRecords.add(record);
                 }
             }
 
-            if (keyExists) {
-                try (var csvPrinter = createNewCsvPrinter(CSV_HEADERS, csvPath)) {
-                    csvPrinter.flush();
-                    csvPrinter.printRecords(updatedRecords);
-                }
+            if (recordExists) {
+                writeNewCsv(updatedRecords);
             }
         } catch (IOException e) {
-            throw new RepositoryException("Failed to invalidate key in Single Use Key CSV", e);
+            throw new RepositoryException("Failed to invalidate record in Single Use Key CSV", e);
         } finally {
             lock.unlock();
         }
     }
 
+    /**
+     * Determines if the specific record should be invalidated
+     * @param key the access key, may be null if we want a time-based eval
+     * @param record the csv record
+     * @return true if the record should be invalidated
+     */
+    private boolean recordShouldBeInvalidated(String key, CSVRecord record) {
+        var currentTime = System.currentTimeMillis();
```

Comment:
Minor, but we can move this into the conditional to make it clearer that the time is only needed when key is null